### PR TITLE
Add error handling

### PR DESCRIPTION
**Pull request by mhabzda**

### CODE REVIEW

File: deezer/src/main/java/agency/tango/skald/deezer/player/DeezerPlayer.java [hunk 1]
```diff
@@ -172,22 +183,17 @@ private void play(SkaldPlaylist skaldPlaylist, SkaldOperationCallback operationC
...
-    operationCallback.onSuccess();
   }
 
-  private void playPlaylist(SkaldOperationCallback operationCallback, long playlistId,
-      PlaylistPlayer playlistPlayer) {
+  private void playPlaylist(long playlistId, PlaylistPlayer playlistPlayer) {
     playlistPlayer.playPlaylist(playlistId);
-    notifyLoadingEvent();
-    operationCallback.onSuccess();
+
```
Comment (bezmian) [inline deezer/src/main/java/agency/tango/skald/deezer/player/DeezerPlayer.java]: why is enter here?

File: deezer/src/main/java/agency/tango/skald/deezer/player/PlayerListener.java [hunk 2]
```diff
@@ -69,14 +73,16 @@ public void onResult(Object result, Object requestId) {
       @Override
       public void onUnparsedResult(String requestResponse, Object requestId) {
         for (OnPlaybackListener onPlaybackListener : onPlaybackListeners) {
-          onPlaybackListener.onError(new PlaybackError("Cannot get track info"));
+          onPlaybackListener.onError(
+              new PlaybackError(new IllegalStateException(
```
Comment (bezmian) [inline deezer/src/main/java/agency/tango/skald/deezer/player/PlayerListener.java]: Czemu IllegalStateException akurat?

File: spotify/src/main/java/agency/tango/skald/spotify/player/SkaldSpotifyPlayer.java [hunk 3]
```diff
@@ -61,6 +63,9 @@ public void onInitialized(final SpotifyPlayer spotifyPlayer) {
           @Override
           public void onError(Throwable throwable) {
             Log.e(TAG, "Could not initialize player", throwable);
+            onErrorListener.onError(
+                new SpotifyPlayerInitializationException("Could not initialize spotify player",
```
Comment (bezmian) [inline spotify/src/main/java/agency/tango/skald/spotify/player/SkaldSpotifyPlayer.java]: Rename to PlayerInitializationException

File: spotify/src/main/java/agency/tango/skald/spotify/player/callbacks/SpotifyConnectionStateCallback.java [hunk 4]
```diff
@@ -55,27 +60,32 @@ public void onLoggedOut() {
...
+          });
+    } else {
+      onErrorListener.onError(new SpotifyException(error));
+    }
   }
 
   @Override
   public void onTemporaryError() {
-
+    onErrorListener.onError(new Exception("Temporary connection error in Spotify"));
```
Comment (bezmian) [inline spotify/src/main/java/agency/tango/skald/spotify/player/callbacks/SpotifyConnectionStateCallback.java]: SpotifyException?

Comment (mhabzda) [reply spotify/src/main/java/agency/tango/skald/spotify/player/callbacks/SpotifyConnectionStateCallback.java]: In SpotifyException we need Error(Spotify class) and here we don't get any. We do not know anything specific about what was wrong that's why I'm using here basic Exception. Is it better to create TemporarySpotifyException?

**Comment by bezmian** [comment]
LGTM

**Pull request merged**